src/lib/CONFIG/config.cpp

```diff
@@ -61,61 +117,169 @@ void TxConfig::Load()
     ESP_ERROR_CHECK( err );
     ESP_ERROR_CHECK(nvs_open("ELRS", NVS_READWRITE, &handle));
 
-    // read version field
-    if(nvs_get_u32(handle, "tx_version", &m_config.version) != ESP_ERR_NVS_NOT_FOUND
-        && m_config.version == (TX_CONFIG_VERSION | TX_CONFIG_MAGIC))
+    // Try to load the version and make sure it is a TX config
+    uint32_t version = 0;
+    if (nvs_get_u32(handle, "tx_version", &version) == ESP_OK && ((version & CONFIG_MAGIC_MASK) == TX_CONFIG_MAGIC))
+        version = version & ~CONFIG_MAGIC_MASK;
+    DBGLN("Config version %u", version);
+
+    // Can upgrade from any version 5 to current
+    if (version < 5)
     {
-        DBGLN("Found version %u config", TX_CONFIG_VERSION);
-        uint32_t value;
-        nvs_get_u32(handle, "vtx", &value);
-        m_config.vtxBand = value >> 24;
-        m_config.vtxChannel = value >> 16;
-        m_config.vtxPower = value >> 8;
-        m_config.vtxPitmode = value;
+        SetDefaults(true);
+        return;
+    }
 
-        nvs_get_u32(handle, "fan", &value);
-        m_config.fanMode = value;
+    SetDefaults(false);
 
-        nvs_get_u32(handle, "motion", &value);
-        m_config.motionMode = value;
+    uint32_t value;
+    uint8_t value8;
+    // vtx (v5)
+    nvs_get_u32(handle, "vtx", &value);
```

Comment:
Shouldn't this have an if around it and only set the value if it returns ESP_OK?

Reply:
I initially had it so every one checked the return value, but the code only runs if the version is 5 or greater. Version 5 config is guaranteed to have these fields and we never checked their return value in the past. The two I did add the check for may or may not be there even if it is version 5 so that's why they got return value checks.

If you think it is now important to check the return values of all the guaranteed items as well, let me know and I can add them.

Reply:
Ok then.

Reply:
I'm happy to change it if you think it is better, I was on the fence about it and made it this way because that was how the current code did it. It would be more correct and potentially safer, but I was just unsure if it was necessary or would ever not return ESP_OK.

Reply:
I was just assumed that if it was not in the nvs then the value will be whatever the variable is set to previously, and as it's a stack variable it could be any random value!

Reply:
I added value = 0 before the call so it gets the correct default.